lib/bugsnag/cleaner.rb

```diff
@@ -29,10 +29,19 @@ def traverse_object(obj, seen, scope)
       when Hash
         clean_hash = {}
         obj.each do |k,v|
-          if filters_match_deeply?(k, scope)
-            clean_hash[k] = FILTERED
-          else
-            clean_hash[k] = traverse_object(v, seen, [scope, k].compact.join('.'))
+          begin
+            if filters_match_deeply?(k, scope)
+              clean_hash[k] = FILTERED
+            else
+              clean_hash[k] = traverse_object(v, seen, [scope, k].compact.join('.'))
+            end
+          # If we get an error here, we assume the key needs to be filtered
+          # to avoid leaking things we shouldn't. We also remove the key itself
+          # because it may cause issues later e.g. when being converted to JSON
+          rescue StandardError
+            clean_hash[RAISED] = FILTERED
+          rescue SystemStackError
+            clean_hash[RECURSION] = FILTERED
```

Comment:
Reusing the same key here does mean we'd overwrite it if more than one key throws, but I think that's unlikely to happen realistically and I don't think you'd get any useful information out of `[RAISED 1]... [RAISED 2]` if we made them unique